librz/bin/format/elf/elf_symbols.c

```diff
@@ -512,7 +515,19 @@ Elf_(Word) Elf_(rz_bin_elf_get_number_of_dynamic_symbols)(RZ_NONNULL ELFOBJ *bin
 		return result;
 	}
 
-	return get_number_of_symbols_from_heuristic(bin);
+	result = get_number_of_symbols_from_section(bin);
+	if (result) {
+		RZ_LOG_WARN("The number of dynamic symbols is deduced from section information.\n");
```

Comment:
This by itself doesn't say much when appearing to the user. Before the `result = get_number_of_symbols_from_section(bin);` line I would do something like
```c
RZ_LOG_WARN("Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...\n");
```

And then in the end if all heuristics have failed:
```c
RZ_LOG_WARN("Failed to determine the number of dynamic symbols from heuristics.\n");
```
(These log strings are just examples, you will probably know better how to formulate them accurately here)

Reply:
i agree on this one.

Reply:
Okay ty